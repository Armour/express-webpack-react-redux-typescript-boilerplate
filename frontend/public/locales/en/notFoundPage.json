{
  "title": "404 Page Not Found"
}
