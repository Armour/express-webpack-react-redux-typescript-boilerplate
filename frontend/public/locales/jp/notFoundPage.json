{
  "title": "404 ページが見つかりません"
}
