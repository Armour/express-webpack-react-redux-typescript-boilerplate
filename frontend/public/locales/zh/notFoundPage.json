{
  "title": "404 页面不存在"
}
