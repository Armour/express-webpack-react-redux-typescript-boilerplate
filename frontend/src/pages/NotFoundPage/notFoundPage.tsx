import React, { Fragment } from 'react';
import { Translation } from 'react-i18next';

const styles = require('./notFoundPage.scss');

const notFoundImageList = [
  '404.gif',
  '404-1.jpeg',
  '404-2.jpeg',
  '404.jpg',
];

interface INotFoundPageState {
  imageId: number;
}

export class NotFoundPage extends React.Component<{}, INotFoundPageState> {
  constructor(props: {}) {
    super(props);
    this.state = { imageId: this.getRandomInt(0, notFoundImageList.length) };
  }

  public getRandomInt = (min: number, max: number) => {
    min = Math.ceil(min);
    max = Math.floor(max);
    return Math.floor(Math.random() * (max - min)) + min; // The maximum is exclusive and the minimum is inclusive
  }

  public render() {
    return (
      <Translation ns='notFoundPage'>
        {(t) => (
          <Fragment>
            <h1 className='page-title'>{t('title')}</h1>
            <img className={styles['not-found-img']} src={require('./assets/images/' + notFoundImageList[this.state.imageId])} alt='not-found-img' height='550px' width='750px' />
          </Fragment>
        )}
      </Translation>
    );
  }
}

export default NotFoundPage;
