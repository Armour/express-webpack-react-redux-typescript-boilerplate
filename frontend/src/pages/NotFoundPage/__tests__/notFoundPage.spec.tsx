import React from 'react';
import TestRenderer from 'react-test-renderer';

import { NotFoundPage } from '../notFoundPage';

describe('NotFoundPage', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <NotFoundPage />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
