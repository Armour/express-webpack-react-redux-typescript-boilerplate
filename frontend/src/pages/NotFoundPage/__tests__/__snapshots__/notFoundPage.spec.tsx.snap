// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage should renders correctly 1`] = `
Array [
  <h1
    className="page-title"
  >
    title
  </h1>,
  <img
    alt="not-found-img"
    className="not-found-img"
    height="550px"
    src={
      Object {
        "default": "test-file-stub",
      }
    }
    width="750px"
  />,
]
`;
